### PR TITLE
239/app instance resources factor out

### PR DESCRIPTION
**Pull request by pyphilia**

### CODE REVIEW

**Comment by juancarlosfarah** [comment]
@pyphilia, I think this has some commits in the other PRs. Let's merge those first and refactor.

File: src/config/messages.js [hunk 1]
```diff
@@ -50,6 +50,9 @@ const SUCCESS_CLEARING_USER_INPUT_MESSAGE =
 const CONNECTION_MESSAGE_HEADER = 'Connection Status';
 const CONNECTION_OFFLINE_MESSAGE = 'You are offline.';
 const CONNECTION_ONLINE_MESSAGE = 'You are online.';
+const ERROR_SIGNING_IN = 'There was an error logging in the user';
```
Comment (juancarlosfarah) [inline src/config/messages.js:53]: Same changes as in backend file.

File: src/config/messages.js [hunk 2]
```diff
@@ -50,6 +50,9 @@ const SUCCESS_CLEARING_USER_INPUT_MESSAGE =
...
+const ERROR_SIGNING_OUT = 'There was an error logging out the user';
```
Comment (juancarlosfarah) [inline src/config/messages.js:54]: Same changes as in backend file.

File: src/config/messages.js [hunk 3]
```diff
@@ -50,6 +50,9 @@ const SUCCESS_CLEARING_USER_INPUT_MESSAGE =
...
+const ERROR_GETTING_AUTHENTICATED = 'There was an error getting authenticated';
```
Comment (juancarlosfarah) [inline src/config/messages.js:55]: Same changes as in backend file.

File: src/components/common/MediaCard.js [hunk 4]
```diff
@@ -87,7 +87,7 @@ export const MediaCard = props => {
 
       {showActions && (
         <CardActions disableSpacing>
-          <ClearButton id={id} />
+          <ClearButton spaceId={id} />
```
Comment (juancarlosfarah) [inline src/components/common/MediaCard.js:90]: Why this change?

File: src/reducers/authenticationReducer.js [hunk 5]
```diff
@@ -16,22 +21,32 @@ import {
...
+  current: {
     activity: List(),
-    lang: DEFAULT_LANGUAGE,
-    developerMode: DEFAULT_DEVELOPER_MODE,
-    geolocationEnabled: DEFAULT_GEOLOCATION_ENABLED,
-  }),
+    folder: null,
+  },
+  authenticated: false,
+  user: Map(DEFAULT_USER),
```
Comment (juancarlosfarah) [inline src/reducers/authenticationReducer.js:49]: I saw somewhere else that the `id` for the `user` is inside the user as `userId`. I would expect this to be inside the `user` object as `id`. Something like `userId = user.id`. What do you think?

File: src/components/space/ExportButton.js [hunk 6]
```diff
@@ -46,11 +47,19 @@ class ExportButton extends Component {
     );
   }
 }
+
+const mapStateToProps = ({ authentication }) => ({
+  userId: authentication.getIn(['user', 'userId']),
```
Comment (juancarlosfarah) [inline src/components/space/ExportButton.js:52]: Here I see that inside `user` we are saving the ID as `userId`. I would've imagined that inside `user`, the user ID would be simply `id`.

File: src/components/signin/SignInScreen.js [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+    } = this.props;
+    const { username, usernameIsEmpty } = this.state;
+
+    return (
+      <Main id={SIGN_IN_MAIN_ID} style={{ background: main }} fullScreen>
+        <FormControl>
+          <img
+            src={logo}
+            alt="graasp logo"
+            width="95%"
```
Comment (juancarlosfarah) [inline src/components/signin/SignInScreen.js:162]: Can we move the width and style to `styles` object?

File: src/components/signin/SignInScreen.js [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+      },
+    } = this.props;
+    const { username, usernameIsEmpty } = this.state;
+
+    return (
+      <Main id={SIGN_IN_MAIN_ID} style={{ background: main }} fullScreen>
+        <FormControl>
+          <img
+            src={logo}
+            alt="graasp logo"
```
Comment (juancarlosfarah) [inline src/components/signin/SignInScreen.js:161]: You should translate `alt` tags too.

File: src/components/phase/PhaseApp.js [hunk 9]
```diff
@@ -294,15 +294,20 @@ class PhaseApp extends Component {
...
 
-  geolocation: User.getIn(['current', 'geolocation']),
-  geolocationEnabled: User.getIn(['current', 'geolocationEnabled']),
+  geolocation: authentication.getIn(['user', 'settings', 'geolocation']),
+  geolocationEnabled: authentication.getIn([
+    'user',
+    'settings',
+    'geolocationEnabled',
+  ]),
+  userId: authentication.getIn(['user', 'userId']),
```
Comment (juancarlosfarah) [inline src/components/phase/PhaseApp.js:310]: I would expect that userId be name `id` within the `user` object:

```
userId: authentication.getIn(['user', 'id']),
```

File: src/components/common/MainMenu.test.js [hunk 10]
```diff
@@ -15,18 +15,19 @@ import {
   DASHBOARD_PATH,
 } from '../../config/paths';
 
-// offline: 2, online: 5
 const MENUITEM_OFFLINE_NUMBER = 2;
-const MENUITEM_OFFLINE_ONLINE_COUNT = 8;
+const MENUITEM_OFFLINE_ONLINE_COUNT = 9;
 
-const createMainMenuProps = (developerMode, path) => {
+const createMainMenuProps = (developerMode, path, authenticated = true) => {
```
Comment (juancarlosfarah) [inline src/components/common/MainMenu.test.js:21]: Do we want to default authenticated to `true`?

File: src/actions/space.js [hunk 11]
```diff
@@ -304,13 +305,17 @@ const clearUserInput = async ({ id }) => async dispatch => {
...
+            spaceId,
+            userId,
+          });
         }
       }
     );
 
     // listen for response from backend
     window.ipcRenderer.once(CLEARED_USER_INPUT_CHANNEL, (event, response) => {
+      console.log('response', response);
```
Comment (juancarlosfarah) [inline src/actions/space.js:318]: Remove this or add a `debug` message instead.

File: public/app/listeners/signIn.js [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+const { USERS_COLLECTION } = require('../db');
+const { SIGN_IN_CHANNEL } = require('../config/channels');
+const {
+  DEFAULT_USER,
+  ANONYMOUS_USERNAME,
+  AUTHENTICATED,
+} = require('../config/config');
+
+const createNewUser = (username, createdAt, anonymous = false) => {
+  const userId = ObjectId().str;
```
Comment (juancarlosfarah) [inline public/app/listeners/signIn.js:12]: This should be `id`, right? If it's inside the `user`.

File: public/app/listeners/signOut.js [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+  try {
+    db.set('user.authenticated', false).write();
+
+    // update user in users collection
+    const user = db.get('user').value();
+    db.get('users')
+      .find({ userId: user.userId })
+      .assign(user)
+      .write();
+
```
Comment (juancarlosfarah) [inline public/app/listeners/signOut.js:15]: Maybe a `// todo` for here would be to save the length of the session in an analytics collection (for general desktop based analytics).

File: src/components/common/MediaCard.js (hunk 4 above)
Comment (pyphilia) [reply src/components/common/MediaCard.js:90]: ClearButton was defined to take spaceId as prop, so it was previously a bug. I'll also change the other buttons so that they all take `spaceId` and not `id`.
